opentelemetry-sdk/src/opentelemetry/sdk/_metrics/_internal/aggregation.py

```diff
@@ -192,28 +204,50 @@ def collect(self) -> Optional[Sum]:
                 self._value = 0
                 self._start_time_unix_nano = now + 1
 
-            return Sum(
-                aggregation_temporality=AggregationTemporality.DELTA,
-                is_monotonic=self._instrument_is_monotonic,
-                start_time_unix_nano=start_time_unix_nano,
-                time_unix_nano=now,
-                value=value,
-            )
+        else:
 
-        with self._lock:
-            if self._value is None:
-                return None
-            value = self._value
-            self._value = None
+            with self._lock:
+                if self._value is None:
+                    return None
+                value = self._value
+                self._value = None
+                start_time_unix_nano = self._start_time_unix_nano
 
-        return Sum(
-            aggregation_temporality=AggregationTemporality.CUMULATIVE,
-            is_monotonic=self._instrument_is_monotonic,
-            start_time_unix_nano=self._start_time_unix_nano,
+        current_point = NumberDataPoint(
+            attributes=self._attributes,
+            start_time_unix_nano=start_time_unix_nano,
             time_unix_nano=now,
             value=value,
         )
 
+        if self._previous_point is None:
+            self._previous_point = current_point
```

Comment:
I don't see where `self._previous_point` is reset AFTER the first iteration. It looks this only happens once which would be a bug

Reply:
Fixed